WcaOnRails/app/jobs/registration_reminder_job.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+class RegistrationReminderJob < SingletonApplicationJob
+  queue_as :default
+
+  def should_send_reminder(competition)
+    competition.registration_reminder_sent_at.nil? || competition.registration_reminder_sent_at < 2.days.ago
+  end
+
+  def perform
+    Competition
+      .visible
+      .where("registration_open <= ? AND registration_open >= NOW()", 1.day.after)
+      .includes(bookmarked_competitions: [:user])
+      .select { |c| should_send_reminder(c) }.each do |competition|
+        competition.update_attribute(:registration_reminder_sent_at, Time.now)
```

Comment:
I think we should wrap this + the next few lines in a transaction, so that we never end up in a situation where we've marked the reminder as sent, but haven't actually created the job to send the email.

(edit: upon further reading, it looks like we're using `deliver_now` instead of `deliver_later` here. I think we should use `deliver_later`, so connectivity issues with Amazon SES don't cause us to crash out of this loop. yes, it's weird that a delayed job could create more delayed jobs, but in this case, i think it's a reasonable design decision.)

Reply:
> Also, how can I make the delayed jobs fire in the test? Right now the test is sad that no emails are sent.

I think you should be able to do something similar to https://github.com/thewca/worldcubeassociation.org/blob/master/WcaOnRails/spec/controllers/competitions_controller_spec.rb#L653-L665 for instance.

Reply:
Thanks, done.

Reply:
> Good idea, switched to `deliver_now`.

You mean "switched to `deliver_later`, right? I think the code was already doing `deliver_now`. (It looks like the currently posted code is doing `deliver_now`.)

> That means there's no need for a transaction, right?

I think academically speaking, there's value in a transaction. You could imagine that due to a bug, or just a straight up server crash, we might set `registration_reminder_sent_at`, and then crash before we get around to queuing up the emails to be sent. Or maybe we queue some of them, but not all of them. A transaction will ensure that either all of this happens, or none of it happens.

Reply:
> > Good idea, switched to `deliver_now`.
> 
> ~You mean "switched to `deliver_later`, right? I think the code was already doing `deliver_now`. (It looks like the currently posted code is doing `deliver_now`.)~ EDIT: Nevermind, I think I was looking at old code when I wrote that message.
> 
In any case, yes I did mean `deliver_later` :)

> > That means there's no need for a transaction, right?
> 
> I think academically speaking, there's value in a transaction. You could imagine that due to a bug, or just a straight up server crash, we might set `registration_reminder_sent_at`, and then crash before we get around to queuing up the emails to be sent. Or maybe we queue some of them, but not all of them. A transaction will ensure that either all of this happens, or none of it happens.

Got it, thanks.  Done.